src/Build/Graph/ProjectGraph.cs

```diff
@@ -11,28 +10,45 @@ namespace Microsoft.Build.Graph
     /// <summary>
     /// Represents a graph of evaluated projects.
     /// </summary>
-    internal sealed class ProjectGraph
+    public sealed class ProjectGraph
     {
         private readonly List<ProjectGraphNode> _projectNodes = new List<ProjectGraphNode>();
 
+        // TODO: We probably want to mirror all relevant constructor overloads from Project
+
         /// <summary>
         /// Constructs a graph starting from the given project file.
         /// </summary>
         /// <param name="entryProjectFile">The project file to use as the entry point in constructing the graph</param>
         /// <exception cref="InvalidProjectFileException">If the evaluation of any project in the graph fails.</exception>
         public ProjectGraph(string entryProjectFile)
+            : this(entryProjectFile, ProjectCollection.GlobalProjectCollection, null, null)
         {
-            var graphNode =
-                new ProjectGraphNode(ProjectCollection.GlobalProjectCollection.LoadProject(entryProjectFile));
-            _projectNodes.Add(graphNode);
         }
 
         /// <summary>
-        /// Constructs a graph starting from the given project files.
+        /// Constructs a graph starting from the given project file.
         /// </summary>
-        /// <param name="entryProjectFiles">The project files to use as the entry points in constructing the graph</param>
+        /// <param name="entryProjectFile">The project file to use as the entry point in constructing the graph</param>
+        /// <param name="globalProperties">The global properties to use for all projects. May be null, in which case the containing project collection's global properties will be used.</param>
+        /// <param name="projectCollection">The collection with which all projects in the graph should be associated. May not be null.</param>
+        /// <param name="toolsVersion">The tools version. May be null.</param>
         /// <exception cref="InvalidProjectFileException">If the evaluation of any project in the graph fails.</exception>
-        public ProjectGraph(IEnumerable<string> entryProjectFiles) { }
```

Comment:
I removed this multi-project ctor as it was more of a theoretical scenario not based in a specific concrete use case, so I'm removing it for now to simplify implementation.

Anyway a metaproj type thing could solve this multi-project entry point anyway if it's really needed.